scio-core/src/main/scala/com/spotify/scio/sql/Sql.scala

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.sql
+
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.schemas._
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.extensions.sql.{BeamSqlTable, SqlTransform}
+import org.apache.beam.sdk.values._
+import org.apache.beam.sdk.schemas.{SchemaCoder, Schema => BSchema}
+import com.spotify.scio.util.ScioUtil
+import org.apache.beam.sdk.extensions.sql.impl.rel.BeamRelNode
+import org.apache.beam.sdk.extensions.sql.impl.{BeamSqlEnv, ParseException}
+import org.apache.beam.sdk.extensions.sql.impl.schema.BaseBeamTable
+import org.apache.beam.sdk.extensions.sql.impl.utils.CalciteUtils
+
+import scala.language.experimental.macros
+import scala.util.Try
+import scala.collection.JavaConverters._
+
+final case class Query[A, B](query: String, udfs: List[Udf] = Nil)
```

Comment:
Would it be possible to also support a TupleTag with a default value here ? 

```scala
final case class Query[A, B](query: String, 
                             aTag: TupleTag[A] = new TupleTag[A]("SCOLLECTION"), 
                             udfs: List[Udf] = Nil)
```

That way we get rid of the awkward `PCOLLECTION` in queries which does not really make sense in Scio

Reply:
Yup, good idea! also will allow uses to name their tables